prov/efa/src/rdm/efa_rdm_pke_rta.c

```diff
@@ -415,10 +417,16 @@ int efa_rdm_pke_proc_fetch_rta(struct efa_rdm_pke *pkt_entry)
 	}
 
 	err = efa_rdm_ope_post_send_or_queue(rxe, EFA_RDM_ATOMRSP_PKT);
-	if (OFI_UNLIKELY(err))
-		efa_rdm_rxe_handle_error(rxe, -err, FI_EFA_ERR_PKT_POST);
 
 	efa_rdm_pke_release_rx(pkt_entry);
+
+	if (OFI_UNLIKELY(err)) {
```

Comment:
Why make this change?

Reply:
This is an internal operation, so it should generate an eq error and release rxe. The old code is wrong here. `efa_rdm_rxe_handle_error` doesn't release the rxe and may write cq error (which this PR will fix as well)

Reply:
But you fixed the behavior of `efa_rdm_rxe_handle_error`. And the rxe here is allocated by `efa_rdm_pke_alloc_rta_rxe` which does set the `EFA_RDM_OPE_INTERNAL` flag

Reply:
I slightly prefer to use a shortcut here to avoid involving unexpected dependencies in `rxe_handle_error`. And it is consistent with the similar code paths in `rtr`, `rtw` . If we want to use `efa_rdm_rxe_handle_error`, we should change them all to be consistent, which can be a follow-up PR.

Reply:
OK, I think there are many resource leak of `pkt_entry` in the rta path. So I'd rather make all such cleanup in a follow up work. I will revert to use `efa_rdm_rxe_handle_error` here but still release the `rxe` because it was a leak anyway